hippod/api_object_post.py

```diff
@@ -290,6 +290,34 @@ def object_index_initial_add_major(app, sha_major, xobj):
     object_index_update(app, object_index)
 
 
+def check_anchor(app, xobj):
+    anchor = False
+    achievements = xobj['achievements']
+    for achievement in achievements:
+        if 'anchor' in achievement:
+            anchor = True
+    return anchor
+
+
+def update_lifetime(app, sha_major, sha_minor, anchor):
+    conf = app['CONF']
+    garbage = hippod.garbage_handler_container_achievements.GHContainerAchievements()
+    ok, subcontainer = hippod.api_shared.read_subcont_obj_by_id(app, sha_major, sha_minor)
+    if not ok:
+        log.error("can not read subcontainer by id '{}/{}'".format(sha_major, sha_minor))
+    default_lifetime = garbage.convert_lifetime(conf.achievements_validity_lifetime.achievements)
+    anchor_lifetime = garbage.convert_lifetime(conf.achievements_validity_lifetime.achievements_anchored)
+    if not anchor and subcontainer['lifetime-leftover'] < default_lifetime:
```

Comment:
Can this be easier to understand? Why not, this looks a little bit more complicated but is better understandable:

```
    if anchor:
        subcontainer['lifetime-leftover'] = anchor_lifetime
    else:
         # no anchor
        #  update the time to default if small (just a example text)
        if subcontainer['lifetime-leftover'] < default_lifetime:
                subcontainer['lifetime-leftover'] = default_lifetime
````

The question I have here is: why is the lifetime update here. Is this line not always true? Should it be `if subcontainer['lifetime-leftover'] > default_lifetime:`